his_checker.py

```diff
@@ -0,0 +1,173 @@
+
+def process_statements(file):
+
+    """Process number of statements in a function"""
+
+    global METRIC_FAIL
+    total_stmt = 0
+    complexity = "pmccabe -c "
+    lines = os.popen(complexity + str(file))
+    lines = lines.read()
+    sline = lines.split('\n')
+
+    for num in range(len(sline)-1):
+        fields = sline[num].split('\t')
+        num_statements = int(fields[2])
+        total_stmt += num_statements
+        if num_statements > STATEMENTS_TRESHOLD:
+            function = fields[5].split(':')
+            print(str(function[1]) + " exceeded the # of statements allowed with "
+                  + fields[2] + " statements (max. " + str(STATEMENTS_TRESHOLD) +
+                  ")" + " at " + str(function[0]))
+            METRIC_FAIL = 1
+    return total_stmt
+
+def process_comments(file, total_stmt):
+
+    """Process comment density in a file"""
+
+    global METRIC_FAIL
+    lines = os.popen("pygount  --format=cloc-xml " + str(file))
+    lines = lines.read()
+    sline = lines.split(' ')
+    for num in range(len(sline)-1):
+        if sline[num].startswith("comment"):
+            comf = sline[num].split("=")
+            comf = int(comf[1].strip('"'))/total_stmt
+            if comf < 0.2:
+                print("Comment density (comments/statements) must be bigger \
+than 0.2! Current value is " + str(comf))
+                METRIC_FAIL = 1
+            break
+
+def process_goto(file):
+
+    """Process number of go to statements"""
+
+    global METRIC_FAIL
+    qmcalc = "qmcalc "
+    lines = os.popen(qmcalc + str(file))
+    lines = lines.read()
+    sline = lines.split('\n')
+
+    for num in range(len(sline)-1):
+        fields = sline[num].split('\t')
+        if int(fields[18]) > GOTO_TRESHOLD:
+            print("Illegal goto expression in a function in" + file)
+            METRIC_FAIL = 1
+
+def process_calls(file):
+
+    """Process number of calls inside a function"""
+
+    global METRIC_FAIL
+    cflow = "cflow -l --depth=2 "
+    if file.endswith(".c"):
+        lines = os.popen(cflow + str(file))
+        lines = lines.read()
+        sline = lines.split('\n')
+        call_counter = 0
+
+        for num in range(len(sline)-1):
+            fields = sline[num].split('\t')
+            fields = fields[0].split('}')
+            fields[0] = fields[0].lstrip("{")
+            fields[0] = fields[0].lstrip()
+            fields[1] = fields[1].lstrip()
+
+            if (int(fields[0]) == 0 and call_counter > CALL_TRESHOLD):
+                print("With " + str(call_counter) + " calls (of a maximum "
+                      + str(CALL_TRESHOLD) + "): " + parent_func)
+                METRIC_FAIL = 1
+
+            if int(fields[0]) == 0:
+                parent_func = fields[1]
+                call_counter = 0
+
+            if int(fields[0]) > 0:
+                call_counter = call_counter + 1
+
+def process_callees(file):
+
+    """Process number of times a function is called in a file"""
```

Comment:
This should be the number of different functions that call a given function.

Reply:
The question is, the scope of this metric should be the whole project? Or the calls for a function within a file?
The official HIS document comments this metric saying "The range 1 to 5 only makes sense when the entire system is analysed. With subsystems and libraries 0 to 5 is permissible. For the complete system there is the exception of main(), since this will only be called by the Start-up-Code."